lib-gphotos/client.go

```diff
@@ -57,31 +56,97 @@ func NewClient(oauthHTTPClient *http.Client, maybeToken ...*oauth2.Token) (*Clie
 }
 
 // GetUploadToken sends the media and returns the UploadToken.
-func (client *Client) GetUploadToken(r io.Reader, filename string) (token string, err error) {
-	req, err := http.NewRequest("POST", fmt.Sprintf("%s/%s/uploads", basePath, apiVersion), r)
+// The uploadURL parameter will be updated
+func (c *Client) GetUploadToken(r io.ReadSeeker, filename string, fileSize int64, uploadURL *string) (uploadToken string, err error) {
```

Comment:
I'm afraid that this is a breaking change. It's not backwards compatible... what about implementing a new method `ResumableUploadToken`? This new method could implement the new contract and we can deprecate the old one... WDYT?

Reply:
This is indeed a breaking change, but I'm not sure `GetUploadToken()` was meant to be public as AFAIK it is only used in `UploadFile()`.

Anyway, I can add a new method as that cannot really be confirmed.

Reply:
That should be resolved now.

Reply:
You're right, `GetUploadToken` should be non-exposed method, but now It's not possible to know if anybody is using it... So I'd suggest a new method for the resumable part.

Reply:
Hi @pdecat

I think we can continue as you approached, so let's go for v2 with breaking changes... This will allow us to rethink / refactor the API in a more convenient way...

Reply:
Backward compatibility is maintained now, don't you think?

Reply:
I don't think so, the `GetUploadToken` signature is different. But I agree with you and we can make a breaking change, this will allow us to make things differnt and fix some methods calls... So forget my previous comment, let's publish v2 of this module, so we are free to break API contract!

Reply:
Oh, indeed, I believe I can fix that too, I just forgot some bits.

Reply:
Note: I'll be AFK for 2 weeks, feel free to make changes if you need to.

Reply:
I've restored the original implementation of `GetUploadToken` and `UploadFile` to maintain backward compatibility and made the new implementation completely independent.

Also, the new `getUploadTokenResumable` is private.